rct229/ruleset_functions/get_building_scc_skylight_roof_ratios_dict.py

```diff
@@ -0,0 +1,66 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.ruleset_functions.get_opaque_surface_type import OpaqueSurfaceType as OST
+from rct229.ruleset_functions.get_opaque_surface_type import get_opaque_surface_type
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+DOOR = schema_enums["SubsurfaceClassificationType"].DOOR.name
+
+
+def get_building_scc_skylight_roof_ratios_dict(climate_zone, building):
+    # required fields for this function are coming from the nested functions
+    scc_dictionary = get_surface_conditioning_category_dict(climate_zone, building)
+    total_res_roof_area = 0.0
+    total_res_skylight_area = 0.0
+    total_nonres_roof_area = 0.0
+    total_nonres_skylight_area = 0.0
+    total_semiheated_roof_area = 0.0
+    total_semiheated_skylight_area = 0.0
+
+    for surface in find_all("building_segments[*].zones[*].surfaces[*]", building):
+        if get_opaque_surface_type(surface) == OST.ROOF:
+            if scc_dictionary[surface["id"]] == SCC.EXTERIOR_RESIDENTIAL:
```

Comment:
Since all the cases require `getattr_(surface, "surface", "area")` and `_helper_calculate_skylight_area(surface)`, it might be a little nicer to call these two functions at the top (at line 27.5).